src/main/java/org/dependencytrack/tasks/repositories/RepositoryMetaAnalyzerTask.java

```diff
@@ -211,6 +213,7 @@ private void analyze(final QueryManager qm, final Component component, final IMe
 
                 if (StringUtils.trimToNull(model.getLatestVersion()) != null) {
                     // Resolution from repository was successful. Update meta model
+                    //FIXME What happens if multiple repositories return a metamodel result with different lastPublishedTimestamps?
```

Comment:
The repository loop is broken out of upon encounter of the first matching repository, via the `break` in line `243`.

Reply:
Yeah, that's what made me wonder. I noticed some components are published in multiple repositories. For example `phpunit`. I came across a third part repo that published an older version of `phpunit`. If DT ends up using that release as `latest`, it might draw the wrong conclusion. Might be safer to analyze all repositories and take the global latest release. But that could be wrong as well if some repo publish a release with some changes/fixes with a higher version number. Not sure how DT handles the suffixes like p1, p2, redhat1, etc.

Reply:
There is no bullet-proof way to determine what the "correct" repository or version is, at least none I am aware of. Hence the current strategy that will work *most of the time*, is to pick the first match. The strategy assumes that authoritative repositories (i.e. Maven Central for Maven) are checked first, which ATM is ensured via the [resolution order](https://github.com/DependencyTrack/dependency-track/blob/2316d55dfdf49879312eef9dbaba34f86609cef4/src/main/java/org/dependencytrack/model/Repository.java#L74-L77).

> Not sure how DT handles the suffixes like p1, p2, redhat1, etc.

It's not possible to handle such things generically. #2826 might help, but only as long as whoever adds these suffixes abides to the ecosystem's de-facto standard notation.